### PR TITLE
Feature: Add Reveal and Modal display options to Donation Form Block v3

### PR DESCRIPTION
**Pull request by JoshuaHungDinh**
## Description

This PR updates the current Donation Form Block v(3) with an additional block control named ```Form Format``` which is a dropdown that allows users to choose from two new display options to render their donation form. 

The two new options are "Reveal" & "Modal". Both display options are compatible with all 3 form templates. Both display options initially hide the users form & only render a single button. If Either of these options are selected an additional control to edit the text within this button becomes available.

If "Reveal" is selected, the form is hidden and when the user clicks the button the form is "revealed" just beneath the button.

If "Modal" is selected, the form is hidden and when the user clicks the button the form is rendered in a modal in the middle of the screen.

## Affects
Donation Form Block v3

## Visuals

https://www.loom.com/share/3b3e92d800e148e5ae06a7dc9e3f4419?sid=ea708d9c-6752-476c-b60d-d907eb7823cb

## Testing Instructions

- Create a v3 form for each form template.
- Create a page and add the Donation Form Block v3 for each template you created.
- Select the block and set the display option to "Reveal" on all three forms.
- Test the additional text control for the 'Open form button'
- While in the block editor verify the forms are hidden and reveal when the button is clicked.
- View the page and verify again that it works properly on the frontend page.
- Repeat steps for the modal option.

## Pre-review Checklist

-   [x] Acceptance criteria satisfied and marked in related issue
-   [ ] Relevant `@unreleased` tags included in DocBlocks
-   [ ] Includes unit tests
-   [ ] Reviewed by the designer (if follows a design)
-   [ ] [Self Review](https://give.gitbook.io/development-manual/devops/github/code-reviews#self-review) of code and UX completed

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1204982550616184

### CODE REVIEW

File: webpack.config.js [hunk 1]
```diff
@@ -25,6 +25,8 @@ module.exports = {
     },
     entry: {
         donationFormBlock: srcPath('DonationForms/Blocks/DonationFormBlock/resources/block.ts'),
+        donationFormBlockApp: srcPath('DonationForms/Blocks/DonationFormBlock/resources/app/index.tsx'),
+        donationFormBlockAppStyle: srcPath('DonationForms/Blocks/DonationFormBlock/resources/app/styles/index.scss'),
```
Comment (pauloiankoski) [inline webpack.config.js:29]: When you import a .scss file into a .tsx, wp-scripts already compile it with the same name as the compiled .js file.
In this case, you'll see a `donationFormBlockApp.css` file with the same content as this `donationFormBlockAppStyle.css`

**Review by pauloiankoski — CHANGES_REQUESTED** [review]
@JoshuaHungDinh I left some specific comments through the code. Other than that, I'd suggest to:
- remove imports of React from react
- replace `<Fragment>` with `<>`
- replace `useState` import from react with `@wordpress/element`

So that we don't bundle react when is not explicitly necessary once WP handles that for us.

File: src/DonationForms/Blocks/DonationFormBlock/Controllers/BlockRenderController.php [hunk 2]
```diff
@@ -39,11 +40,15 @@ public function render(array $attributes)
...
+        $formFormat = $blockAttributes->formFormat ?? '';
+
+        $openFormButton = $blockAttributes->openFormButton ?? '';
+
         /**
          * Note: iframe-resizer uses querySelectorAll so using a data attribute makes the most sense to target.
          * It will also generate a dynamic ID - so when we have multiple embeds on a page there will be no conflict.
          */
-        return "<iframe data-givewp-embed src='$viewUrl' data-givewp-embed-id='$embedId' style='width: 1px;min-width: 100%;border: 0;transition: height 0.25s;'></iframe>";
+        return "<div class='root-data-givewp-embed' data-src='$viewUrl' data-givewp-embed-id='$embedId' data-form-format='$formFormat' data-open-form-button='$openFormButton'></div>";
```
[the remark this answers is not in the document]
Comment (JoshuaHungDinh) [reply src/DonationForms/Blocks/DonationFormBlock/Controllers/BlockRenderController.php]: Resolved: ad9c27590ec13793773d7874bc79b0c896fadb17

File: src/DonationForms/Blocks/DonationFormBlock/resources/editor/Edit.tsx [hunk 3]
```diff
@@ -52,6 +55,40 @@ export default function Edit({clientId, attributes, setAttributes}: BlockEditPro
...
                         )}
                     </PanelRow>
+                    <PanelRow>
+                        <SelectControl
+                            label={__('Form Format', 'give')}
+                            value={formFormat}
+                            options={[
+                                {
+                                    label: __('Full Form', 'give'),
+                                    value: 'full form',
```
[the remark this answers is not in the document]
Comment (JoshuaHungDinh) [reply src/DonationForms/Blocks/DonationFormBlock/resources/editor/Edit.tsx]: Resolved: a4669e00f91df6a608c1757a488a06507f70d4ff

File: webpack.config.js (hunk 1 above)
Comment (JoshuaHungDinh) [reply webpack.config.js]: Ahh ok got it! That makes sense. I re-updated the wp_enqueue_style. Resolved: 0900da3f6ff61389a8c577b1445d6f2b0af98301

File: src/DonationForms/Blocks/DonationFormBlock/block.json [hunk 4]
```diff
@@ -16,6 +16,14 @@
         },
         "blockId": {
             "type": "string"
+        },
+        "formFormat": {
+            "type": "string",
+            "default": "Full Form"
```
[the remark this answers is not in the document]
Comment (JoshuaHungDinh) [reply src/DonationForms/Blocks/DonationFormBlock/block.json]: Resolved: 29a2d3b96c83fe9f08aa5adbaaf75c54cf13f141

**Review by pauloiankoski — APPROVED** [review]

**Review by rickalday — APPROVED** [review]
Passed manual QA tests

**Pull request merged**